js/main.js

```diff
@@ -0,0 +1,67 @@
+let books;
+
+// loader
+function loader() {
+  document.getElementById('view').innerHTML = '';
+  const warning = document.querySelector('.warning');
+  warning.style.visibility = 'hidden';
+  let article = '';
+  books.forEach((book) => {
+    article += `
+            <article>
+             <p class="book-title">${book.title}</p>
+             <p class="author">${book.author}</p>
+             <button class="remove" onclick="whatButton(this)">Remove</button>
+             <hr>
+              </article>
+    `;
+  });
+  document.getElementById('view').innerHTML = article;
+}
+
+if (books === []) {
+  localStorage.setItem('books', JSON.stringify(books));
+} else {
+  books = JSON.parse(localStorage.getItem('books'));
+  loader();
+}
```

Comment:
- [ ] when opening the application with empty localstorage the user receives an error in the console that `books` is null this is because the code is checking for an empty array in this if statement while the `books` variable is actually `undefinded`. if you change this if statement to something like:
```
if (books === undefined)
```
it should work.

Reply:
Done 👍